### PR TITLE
feat(TA-471): add circle config

### PR DESCRIPTION
**Pull request by darren-arbon**
#### Problem
[TA-471](https://techfromsage.atlassian.net/browse/TA-471)

Make this repo align with our other repos and use circle for stuff.

#### Change/Fix
This was the original config for github pages deploy
<table><tr><td>
    
</td></tr></table>

It has been changed to this
<table><tr><td>
    
</td></tr></table>

1. Circle CI config set up to run 4 jobs
	a. Start
    b. Be Kind To Your Colleagues
    c. Lint and test
          This runs on all branch and main builds. Does the linting checks
    d. Push to Github Pages
		  This only runs if the linting passes, and if the branch is main
2. For circle to be able to run `gh-pages` which pushes to the `gh-pages` branch which is used in the above config, circle needed an SSH token that had write permissions, the default ones just have write, so an SSH key was made to work with this.

This has been tested and works fine.

#### Checklist
- [ ] PO review
- [ ] UX review
- [ ] Accessibility review (See: https://github.com/talis/elevate-app/wiki/Accessibility)

### CODE REVIEW

File: .stylelintrc.json [hunk 1]
```diff
@@ -50,6 +50,6 @@
         "severity": "warning"
       }
     ],
-    "import-notation": "less"
+    "import-notation": "string"
```
Comment (darren-arbon) [inline .stylelintrc.json:53]: Whilst this was not causing failures in the linting, less was apparently not a valid value and was causing the npm test step to fail

File: package.json [hunk 2]
```diff
@@ -22,20 +22,21 @@
     "build": "npm-run-all build:*",
     "build:css": "sass scss/:docs/assets/css/ --style=compressed --load-path=node_modules --source-map",
     "build:eleventy": "eleventy",
+    "css": "npm-run-all css-compile css-prefix",
     "css-compile": "sass scss/:docs/assets/css/ --style=compressed --load-path=node_modules --source-map",
     "css-lint": "stylelint scss/",
-    "format:stylelint": "stylelint scss/ --fix",
```
Comment (darren-arbon) [inline package.json:27]: BW: re-ordered these in alpha

File: package.json [hunk 3]
```diff
@@ -22,20 +22,21 @@
...
     "css-prefix": "postcss --replace docs/assets/css/talis.css --use autoprefixer --map",
     "css-purge": "purgecss --keyframes --css docs/assets/css/talis.css --content docs/index.html \"node_modules/bootstrap/js/dist/{util,modal}.js\" --output docs/assets/css/",
-    "css": "npm-run-all css-compile css-prefix",
+    "format:stylelint": "stylelint scss/ --fix",
+    "prepare": "husky install",
+    "publish:githubpages": "gh-pages --dist docs",
```
Comment (darren-arbon) [inline package.json:32]: This is new and needed for the circle config

File: package.json [hunk 4]
```diff
@@ -46,6 +47,7 @@
     "@commitlint/cli": "17.6.1",
     "@commitlint/config-conventional": "17.6.1",
     "autoprefixer": "10.4.14",
+    "gh-pages": "6.1.1",
```
Comment (darren-arbon) [inline package.json:50]: This package allows us to do the release to github pages

**Review by twba-a — APPROVED** [review]
not that i'm expert on this but LGTM.

**Pull request merged**